dask/dataframe/methods.py

```diff
@@ -340,8 +341,9 @@ def cummax_aggregate(x, y):
 def assign(df, *pairs):
     # Only deep copy when updating an element
     # (to avoid modifying the original)
+    # Setitem never modifies an array inplace with pandas 1.4 and up
     pairs = dict(partition(2, pairs))
-    deep = bool(set(pairs) & set(df.columns))
+    deep = bool(set(pairs) & set(df.columns)) and not PANDAS_GT_140
     df = df.copy(deep=bool(deep))
     for name, val in pairs.items():
         df[name] = val
```

Comment:
Why can't we just do `df = df.assign(pairs)`? I would assume this to be handled better by pandas.

Reply:
assign copies the data while ``df[x] = something`` doesn't